### ISSUE TITLE
Unexpected ERR_UseDefViolation error is caused by an erroneous use of `!`.

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29906]
```
        [Fact]
        public void SuppressNullableWarning_Condition()
        {
            var source =
@"class C
{
    static object? F(bool b)
    {
        return (b && G(out var o))!? o : null;
    }
    static bool G(out object o)
    {
        o = new object();
        return true;
    }
}";
            var comp = CreateCompilation(
                new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition },
                parseOptions: TestOptions.Regular8);
            comp.VerifyDiagnostics(
                // (5,16): warning CS8624: The suppression operator (!) can only be applied to reference types.
                //         return (b && G(out var o))!? o : null;
                Diagnostic(ErrorCode.WRN_SuppressionOperatorNotReferenceType, "(b && G(out var o))!").WithLocation(5, 16),
                // (5,38): error CS0165: Use of unassigned local variable 'o'
                //         return (b && G(out var o))!? o : null;
                Diagnostic(ErrorCode.ERR_UseDefViolation, "o").WithArguments("o").WithLocation(5, 38)
                );
        }
```
